src/lazy/expanded/macro_evaluator.rs

```diff
@@ -1239,17 +1239,31 @@ impl<'top> TemplateExpansion<'top> {
 
 #[cfg(test)]
 mod tests {
-    use crate::{v1_1, ElementReader, Int, IonResult, MacroTable, Reader};
+    use crate::{v1_1, ElementReader, Int, IonResult, MacroTable, Reader, Sequence};
 
     /// Reads `input` and `expected` and asserts that their output is Ion-equivalent.
     fn stream_eq<'data>(input: &'data str, expected: &'data str) -> IonResult<()> {
         let mut actual_reader = Reader::new(v1_1::Text, input)?;
         let actual = actual_reader.read_all_elements()?;
 
+        assert_eq_expected(&actual, expected)
+    }
+
+    fn bin_stream_eq(input: &[u8], expected: &str) -> IonResult<()> {
```

Comment:
🗺️ The unit test helper `stream_eq` was hardcoded to expect 1.1 text so each call site didn't have to repeat the 1.1 IVM. For this PR I needed to test reading binary, streams, so I added an analogous `bin_stream_eq`.

Reply:
The diff generated here is not ideal, and that made me wonder which diffing algorithm GitHub uses. Apparently it (like other review tools I'm familiar with) uses the Git default (Myers): https://stackoverflow.com/questions/61769668/whats-the-diff-algorithm-used-by-github

The Patience algorithm would do better here. Here's an analogous example: https://gist.github.com/roryokane/6f9061d3a60c1ba41237

Nothing to do here, I'm just pointing it out so that from now on you can also be slightly dissatisfied when you encounter odd diffs. 

EDIT: One thing you can do is +1 this community discussion? https://github.com/orgs/community/discussions/16782